client.go

```diff
@@ -210,12 +235,21 @@ func newBulkPostRequest(urlStr, apiKey string, ips []string) (*http.Request, err
 		return nil, errors.New("must provide at least one IP")
 	}
 
+	if err := ctx.Err(); err != nil {
+		switch err {
+		case context.DeadlineExceeded:
+			return nil, errors.New("context timeout exceeded")
+		case context.Canceled:
+			return nil, errors.New("context cancelled")
+		}
+	}
```

Comment:
I would simplify the whole block to:

```Go
if err := ctx.Err(); err != nil {
  return nil, err
}
```

I think you'll even end up with the same error messages. 😄

Reply:
No worries, I fixed it.